client-sdk/ts-web/rt/src/address.ts

```diff
@@ -1,18 +1,38 @@
 import * as oasis from '@oasisprotocol/client';
+import * as elliptic from 'elliptic';
+import {Keccak} from 'sha3';
 
 import * as types from './types';
 
-export const V0_SECP256K1_CONTEXT_IDENTIFIER = 'oasis-runtime-sdk/address: secp256k1';
-export const V0_SECP256K1_CONTEXT_VERSION = 0;
+export const V0_SECP256K1ETH_CONTEXT_IDENTIFIER = 'oasis-runtime-sdk/address: secp256k1eth';
+export const V0_SECP256K1ETH_CONTEXT_VERSION = 0;
 export const V0_MULTISIG_CONTEXT_IDENTIFIER = 'oasis-runtime-sdk/address: multisig';
 export const V0_MULTISIG_CONTEXT_VERSION = 0;
 
-export async function fromSecp256k1PublicKey(pk: Uint8Array) {
-    return await oasis.address.fromData(
-        V0_SECP256K1_CONTEXT_IDENTIFIER,
-        V0_SECP256K1_CONTEXT_VERSION,
-        pk,
-    );
+export const ADDRESS_PREFIX = 'oasis';
+
+const SECP256K1 = new elliptic.ec('secp256k1');
+
+export async function fromSigspec(spec: types.SignatureAddressSpec) {
+    if ('ed25519' in spec) {
+        return await oasis.staking.addressFromPublicKey(spec.ed25519);
+    } else if ('secp256k1eth' in spec) {
+        // Use a scheme such that we can compute Secp256k1 addresses from Ethereum
+        // addresses as this makes things more interoperable.
+        const untaggedPk = SECP256K1.keyFromPublic(Array.from(spec.secp256k1eth))
+            .getPublic(false, 'array')
+            .slice(1);
+        const hash = new Keccak(256);
+        hash.update(Buffer.from(untaggedPk));
```

Comment:
I need to go through this process of finding the least bad dependency myself

unless you'd share your findings that led to this one that requires Buffer

Reply:
Oh, no specific reason so this may not be the best choice. Sorry, I'm not familiar with all these different array/buffer representations in jsland.